tests/infra/network.py

```diff
@@ -954,11 +954,11 @@ def find_primary(self, nodes=None, timeout=10, log_capture=None, **kwargs):
                         )
 
             if primary_id is not None:
+                flush_info(logs, log_capture, 0)
```

Comment:
This still potentially flushes in 2 different places with confusingly different line numbers. Can we flush on line 961, after break but before maybe-raising, so we always log from the same line number?

Reply:
Ok, this is a little less verbose, but if it passes I think that's fine.